snuba/migrations/snuba_migrations/outcomes/0003_outcomes_add_category_and_size.py

```diff
@@ -0,0 +1,185 @@
+from typing import Sequence
+
+from snuba.clickhouse.columns import Column, UInt, String, DateTime
+from snuba.clusters.storage_sets import StorageSetKey
+from snuba.migrations import migration, operations
+from snuba.migrations.columns import MigrationModifiers as Modifiers
+
+# TODO: materialized view migration
+old_materialized_view_columns: Sequence[Column[Modifiers]] = [
+    Column("org_id", UInt(64)),
+    Column("project_id", UInt(64)),
+    Column("key_id", UInt(64)),
+    Column("timestamp", DateTime()),
+    Column("outcome", UInt(8)),
+    Column("reason", String()),
+    Column("times_seen", UInt(64)),
+]
+
+new_materialized_view_columns: Sequence[Column[Modifiers]] = [
+    Column("org_id", UInt(64)),
+    Column("project_id", UInt(64)),
+    Column("key_id", UInt(64)),
+    Column("timestamp", DateTime()),
+    Column("outcome", UInt(8)),
+    Column("reason", String()),
+    Column("category", UInt(8)),
+    Column("size", UInt(64)),
+    Column("times_seen", UInt(64)),
+]
+
+
+class Migration(migration.MultiStepMigration):
+    """
+    Adds the http columns defined, with the method and referer coming from the request interface
+    and url materialized from the tags.
+    """
+
+    blocking = False
+
+    def forwards_local(self) -> Sequence[operations.Operation]:
+        return [
+            operations.AddColumn(
+                storage_set=StorageSetKey.OUTCOMES,
+                table_name="outcomes_raw_local",
+                column=Column("size", UInt(32, Modifiers(nullable=True))),
+                after=None,
+            ),
+            operations.AddColumn(
+                storage_set=StorageSetKey.OUTCOMES,
+                table_name="outcomes_raw_local",
+                column=Column("category", UInt(8)),
```

Comment:
Which categories do we have? We can use LowCardinality strings for enum style strings (they are then stored as integers but visible as strings).

Reply:
👍🏼 100%, was getting some weird seg faults when trying low cardinality modifiers so i removed them temporarily, will definitely add them back in.